src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -3098,6 +3326,802 @@ public static Vector128<float> Invoke(Vector128<float> x, Vector128<float> y)
+#endif
+
+            static virtual float IdentityValue => throw new NotSupportedException();
+        }
+
+        /// <summary>MathF.Max(x, y) (but NaNs may not be propagated)</summary>
+        private readonly struct IndexOfMaxOperator : IIndexOfOperator
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static int Invoke(ReadOnlySpan<float> result)
+            {
+                float curMax = result[0];
+                int curIn = 0;
+                if (float.IsNaN(curMax))
+                {
+                    return curIn;
+                }
+
+                for (int i = 1; i < result.Length; i++)
+                {
+                    float current = result[i];
+                    if (float.IsNaN(current))
+                    {
+                        return i;
+                    }
+
+                    if (curMax == current)
+                    {
+                        if (IsNegative(curMax) && !IsNegative(current))
+                        {
+                            curMax = current;
+                            curIn = i;
+                        }
+                    }
+                    else if (current > curMax)
+                    {
+                        curMax = current;
+                        curIn = i;
+                    }
+                }
+
+                return curIn;
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static int Invoke(Vector128<float> result, Vector128<int> maxIndex)
+            {
+                float curMax = float.MinValue;
+                int curIn = 0;
+                for (int i = 0; i < Vector128<float>.Count; i++)
+                {
+                    if (result[i] == curMax && IsNegative(curMax) && !IsNegative(result[i]))
+                    {
+                        curMax = result[i];
+                        curIn = maxIndex[i];
+                    }
+                    else if (result[i] > curMax)
+                    {
+                        curMax = result[i];
+                        curIn = maxIndex[i];
+                    }
+                }
+                return curIn;
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static void Invoke(ref Vector128<float> max, Vector128<float> current, ref Vector128<int> maxIndex, Vector128<int> curIndex)
+            {
+                maxIndex = Vector128.ConditionalSelect(Vector128.Equals(max, current).AsInt32(),
+                    Vector128.ConditionalSelect(IsNegative(max).AsInt32(), curIndex, maxIndex),
+                    Vector128.ConditionalSelect(Vector128.GreaterThan(max, current).AsInt32(), maxIndex, curIndex));
+
+                max = Vector128.ConditionalSelect(Vector128.Equals(max, current),
+                        Vector128.ConditionalSelect(IsNegative(max), current, max),
+                        Vector128.Max(max, current));
```

Comment:
Is this resulting in good codegen? It's a lot of conditionals and some of it is handling cases that are expected to be exceptional.

Reply:
How can I check if the codegen is good? Lets sync on that tomorrow.